File: merge-two-sorted-lists/merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        new_list = ListNode(-1)
        it = new_list
        while l1 and l2:
            if l1.val <= l2.val:
                it.next = l1
                l1 = l1.next
            else:
                it.next = l2
                l2 = l2.next
            it = it.next
        while l1:
            it.next = l1
            l1 = l1.next
            it = it.next
        while l2:
            it.next = l2
            l2 = l2.next
            it = it.next
        return new_list.next

File: merge-two-sorted-lists/merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution 
{
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) 
    {
        // One of the lists is empty
        if (l1 == null)
            return l2;
        if (l2 == null)
            return l1;
        
        // Assign a starting value to head
        ListNode head = null;
        if (l1.val <= l2.val)
        {
            head = l1;
            l1 = l1.next;
        }
        else
        {
            head = l2;
            l2 = l2.next;
        }
        
        // Iterate both lists
        ListNode iterator = head;
        while (l1 != null && l2 != null)
        {
            if (l1.val <= l2.val)
            {
                iterator.next = l1;
                l1 = l1.next;
            }
            else
            {
                iterator.next = l2;
                l2 = l2.next;
            }
            iterator = iterator.next;
        }
        
        // Iterate remaining lists
        while (l1 != null)
        {
            iterator.next = l1;
            l1 = l1.next;
            iterator = iterator.next;
        }
        while (l2 != null)
        {
            iterator.next = l2;
            l2 = l2.next;
            iterator = iterator.next;
        }
        
        return head;
    }
}